Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitUserContentManager.cpp

```diff
@@ -438,6 +513,68 @@ static void testUserContentManagerScriptMessageInWorldReceived(UserScriptMessage
+    g_assert_false(test->registerHandler("msg"));
+
+    test->loadHtml("<html></html>", nullptr);
+    test->waitUntilLoadFinished();
+
+    // Check that the "window.webkit.messageHandlers" namespace exists.
+    GUniqueOutPtr<GError> error;
+    WebKitJavascriptResult* javascriptResult = test->runJavaScriptAndWaitUntilFinished("window.webkit.messageHandlers ? 'y' : 'n';", &error.outPtr());
+    g_assert_nonnull(javascriptResult);
+    g_assert_no_error(error.get());
+    GUniquePtr<char> valueString(WebViewTest::javascriptResultToCString(javascriptResult));
+    g_assert_cmpstr(valueString.get(), ==, "y");
+
+    // Check that the "document.webkit.messageHandlers.msg" namespace exists.
+    javascriptResult = test->runJavaScriptAndWaitUntilFinished("window.webkit.messageHandlers.msg ? 'y' : 'n';", &error.outPtr());
+    g_assert_nonnull(javascriptResult);
+    g_assert_no_error(error.get());
+    valueString.reset(WebViewTest::javascriptResultToCString(javascriptResult));
+    g_assert_cmpstr(valueString.get(), ==, "y");
+
+    valueString.reset(jsc_value_to_string(test->postMessageAndWaitForPromiseResolved("msg", "'Inception'")));
+    g_assert_cmpstr(valueString.get(), ==, "Inception");
+    valueString.reset(jsc_value_to_string(test->postMessageAndWaitForPromiseResolved("msg", "", nullptr, "err msg")));
+    g_assert_cmpstr(valueString.get(), ==, "err msg");
+
+    // Check that the "window.webkit.messageHandlers" namespace doesn't exist in isolated worlds.
+    javascriptResult = test->runJavaScriptInWorldAndWaitUntilFinished("window.webkit.messageHandlers ? 'y' : 'n';", "WebExtensionTestScriptWorld", &error.outPtr());
+    g_assert_null(javascriptResult);
+    g_assert_error(error.get(), WEBKIT_JAVASCRIPT_ERROR, WEBKIT_JAVASCRIPT_ERROR_SCRIPT_FAILED);
+    test->unregisterHandler("msg", nullptr);
+
+    g_assert_true(test->registerHandler("msg", "WebExtensionTestScriptWorld"));
+
+    // Check that the "window.webkit.messageHandlers" namespace exists in the world.
+    javascriptResult = test->runJavaScriptInWorldAndWaitUntilFinished("window.webkit.messageHandlers ? 'y' : 'n';", "WebExtensionTestScriptWorld", &error.outPtr());
+    g_assert_nonnull(javascriptResult);
+    g_assert_no_error(error.get());
+    valueString.reset(WebViewTest::javascriptResultToCString(javascriptResult));
+    g_assert_cmpstr(valueString.get(), ==, "y");
+
+    valueString.reset(jsc_value_to_string(test->postMessageAndWaitForPromiseResolved("msg", "'Inception'", "WebExtensionTestScriptWorld")));
+    g_assert_cmpstr(valueString.get(), ==, "Inception");
+    // Not replying to the script from callback should give us an undefined value.
+    valueString.reset(jsc_value_to_string(test->postMessageAndWaitForPromiseResolved("msg", "", "", "")));
+    g_assert_cmpstr(valueString.get(), ==, "undefined");
+    valueString.reset(jsc_value_to_string(test->postMessageAndWaitForPromiseResolved("msg", "", "WebExtensionTestScriptWorld", "err msg")));
```

Comment:
There is this error in the EWS log:

```
ERROR:/app/webkit/Tools/TestWebKitAPI/Tests/WebKitGLib/TestWebKitUserContentManager.cpp:380:
static void UserScriptMessageTest::runAsyncJavaScriptFinished(GObject*, GAsyncResult*, UserScriptMessageTest*):
assertion failed (test->m_userScriptMessageErrorMessage.get() == error->message): (NULL == "TypeError: Not enough arguments")
```

This is caused because the second argument to the function (empty string) is placed in the JS snippet `window.webkit.messageHandlers.%s.postMessage(%s)` inside the parenthesis, which results in not passing any parameter:

```
window.webkit.messageHandlers.msg.postMessage()
```

For example we could pass `null` or `undefined` instead, to avoid having zero arguments:

```suggestion
    valueString.reset(jsc_value_to_string(test->postMessageAndWaitForPromiseResolved("msg", "undefined", "", "")));
    g_assert_cmpstr(valueString.get(), ==, "undefined");
    valueString.reset(jsc_value_to_string(test->postMessageAndWaitForPromiseResolved("msg", "undefined", "WebExtensionTestScriptWorld", "err msg")));
```

I believe the test case should start passing with this small change.